jerry-core/parser/regexp/re-compiler.c

```diff
@@ -180,7 +186,12 @@ re_insert_into_group (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
 
   qmin = re_ctx_p->current_token.qmin;
   qmax = re_ctx_p->current_token.qmax;
-  JERRY_ASSERT (qmin <= qmax);
+
+  if (qmin > qmax)
```

Comment:
Both `start_opcode` and `end_opcode` are computed before this check but their value will not be used if `qmin > qmax`. Compilers may or may not optimize this but it's safer not to write dead code in the first place. The following sequence might be better:

```c
uint32_t qmin = re_ctx_p->current_token.qmin;
uint32_t qmax = re_ctx_p->current_token.qmax;
if (qmin > qmax)
{
  // ...
}
re_opcode_t start_opcode = re_get_start_opcode_type (re_ctx_p, is_capturable);
re_opcode_t end_opcode = re_get_end_opcode_type (re_ctx_p, is_capturable);
uint32_t start_head_offset_len = re_get_bytecode_length (re_ctx_p->bytecode_ctx_p);
```